bin/spack-install-helper-10-preprocessing

```diff
@@ -8,9 +8,36 @@
 # A100 = 8.0
 # H100 = 9.0
 
+flag_in_list() {
+    for arg in "$@"
+    do
+        if [[ "$arg" == --* ]]; then
+            return 0
+        fi
+    done
+
+    return 1
+}
+
 SPACK_REPO_LIST=()
+SPECS=()
+SPECS_ONLY_DEPENDENCIES=()
+TARGET_MACHINE=
+
+POSSIBLE_TARGETS="daint-gpu daint-mc alps-zen2 alps-a100 alps-mi200 alps-gh200 alps-mi300a"
+DEFAULT_VARIANTS="+openmp +mpi build_type=Release"
+CUDA_EXTERNAL_DEF=""
+ROCM_EXTERNAL_DEF=""
+ROCM_ARCH=""
+CUDA_ARCH=""
+
+if ! flag_in_list "$@"
```

Comment:
Could you leave a comment that this is for backwards compatibility where people are using `spack-install-helper target-machine spec1 spec2`